src/porepy/viz/exporter.py

```diff
@@ -180,6 +182,183 @@ def __init__(
+            # Make some checks
+            if not value.size % grid.num_cells == 0:
+                # This line will raise an error if node or face data is exported.
+                raise ValueError("The data array is not compatible with the grid.")
+
+            # Apply flatten with the order compatible with _toVectorFormat.
+            return value.flatten("F")
+
+        # Convert file_names to a list
+        if isinstance(file_names, str):
+            file_names = [file_names]
+
+        # Consider each file separately.
+        # Procedure has three steps:
+        # 1. Determine dimensionality and type of grid, addressed by the vtu file.
+        # 2. Check whether the vtu file is compatible with the corresponding grids.
+        # 3. Transfer data from vtu to the mixed-dimensional grid.
+        for i, file_name in enumerate(file_names):
+
+            # Read all data from vtu
+            vtu_data = meshio.read(file_name)
+
+            # 1. Step: Determine dimension of the grid associated to vtu file, and
+            # whether the grid corresponds to a subdomain or interface.
+
+            # Allow for automatic detection from the file name.
+            if kwargs.pop("automatic", True):
+                # Assuming grid_dim and is_mortar are among the stored vtu data
+                # one could utilize those, but this does not have to be the case.
+                # Thus, use naming convention of the Exporter for this.
+
+                # Remove ending '.vtu' from file_name, and decompose into pieces
+                # separated by '_'.
+                file_name_pieces = Path(file_name).stem.split("_")
+
+                # If the last two are numbers, the first one denotes the dimension.
+                assert file_name_pieces[-1].isnumeric()
+                dim_pos = -2 if file_name_pieces[-2].isnumeric() else -1
+                dim = int(file_name_pieces[dim_pos])
+
+                # If the key words before are 'mortar', or 'mortar' and 'constant',
+                # the vtu file corresponds to intefaces; otherwise to subdomains.
+                is_subdomain_data = not (
+                    file_name_pieces[dim_pos - 1] == "mortar"
+                    or file_name_pieces[dim_pos - 1] == "constant"
+                    and file_name_pieces[dim_pos - 2] == "mortar"
+                )
+
+            else:
+                # Read from keyword arguments
+
+                # Dimensionality of the grid
+                if "dims" not in kwargs:
+                    raise ValueError(
+                        """Provide dimensionality of the grids associated
+                        to the vtu files."""
+                    )
+                dims = kwargs.pop("dims")
+                assert isinstance(dims, list) or isinstance(dims, int)
+                dim = dims[i] if isinstance(dims, list) else dims
+
+                # Whether the grid is a subdomain or interface
+                if "are_subdomain_data" not in kwargs:
```

Comment:
Any reason not to make `are_subdomain_data` a standard argument.

Reply:
Same as for `dims`.